README.md

```diff
@@ -48,59 +49,168 @@ For .NET Core CLI, use the following command:
+| **Best Practices**| Secure token storage and graceful handling of token expiration.|
+
+To initialize the client for OAuth authentication, specify the credentials explicitly:
+
+```csharp
+var alfred = await AlfredFactory.CreateWithOAuthAsync("username", "password", EnvironmentType.Production);
+```
+
+|HMAC||
+|--|--|
+| **When to Use**    | Best for applications needing individual account access without token management, ensuring secure transactions.|
+| **Benefits**       | Secure request authentication with a unique signature per request.|
+| **Best Practices** | Secure key management, include timestamps to prevent replay attacks, and use a dynamic nonce for each request.|
+
+To initialize the client for HMAC authentication, provide the secret key and public key:
+
+```csharp
+var alfred = await AlfredFactory.CreateWithHmacAsync("secret_key", "public_key", EnvironmentType.Production);
+```
+
+|App API Key||
+|--|--|
+| **When to Use**    | Recommended for server-to-server communication, automated processes, and background services where user interaction is not feasible.|
+| **Benefits**       | Non-interactive yet robust authentication, simplicity, and easy access control.|
+| **Best Practices** | Secure storage of API keys, apply the principle of least privilege, and regularly rotate keys to minimize compromise risks.|
+
+To initialize the client for API key authentication, provide the API key:
+
+```csharp
+var alfred = await AlfredFactory.CreateWithApiKeyAsync("api_key", EnvironmentType.Staging);
+```
+
+# Getting Started
+
+This library covers 6 different domains: 
+- [Tagshelf](#tagshelf-tagshelf-domain)
+- [Deferred Sessions](#deferred-sessions-deferred-sessions-domain)
+- [Files](#files-files-domain)
+- [Jobs](#jobs-jobs-domain)
+- [Data Points](#data-points-data-points-domain)
+- [Account](#account-account-domain)
 
-### Initialize the Client
+## Tagshelf {#tagshelf-domain}
 
-   Begin by creating an instance of the Alfred client using your preferred authentication method along with the desired environment type (Production or Staging). The following examples demonstrate how to initialize the client with different authentication methods:
+The Tagshelf API namespace serves as a dedicated domain within the API for platform status and identity queries. This namespace is tailored to provide essential functionalities related to the operational status and user context of the platform. It encompasses endpoints that are crucial for system health checks, user identity verification, and basic connectivity tests. 
 
-#### Environments description
+### Check Platform Operational Status
+
+This endpoint offers a quick way to check the overall operational status of the platform.
+
+```csharp
+// Authentication
+var alfred = await AlfredFactory.CreateWithOAuthAsync("admin@tagshelf.com", "T@gSh3lf", EnvironmentType.Staging);
```

Comment:
- Remove the admin@tagshelf.com account and its corresponding password, use a placeholder instead.
- `EnviromentType.Staging` I believe it should be `EnviromentType.Production`

Reply:
Both things corrected through out the entire file.